scripts/codeowner_linter.py

```diff
@@ -0,0 +1,281 @@
+    of the CODEOWNERS file.
+
+    Arguments:
+        important_patterns: list(str). List of the important
+            patterns for CODEOWNERS file.
+
+    Returns:
+        bool. Whether the CODEOWNERS "important pattern" check fails.
+    """
+
+    failed = False
+
+    # Check that there are no duplicate elements in the lists.
+    important_patterns_set = set(important_patterns)
+    codeowner_important_paths_set = set(CODEOWNER_IMPORTANT_PATHS)
+    if len(important_patterns_set) != len(important_patterns):
+        python_utils.PRINT(
+            '%s --> Duplicate pattern(s) found in critical rules'
+            ' section.' % CODEOWNER_FILEPATH)
+        failed = True
+    if len(codeowner_important_paths_set) != len(CODEOWNER_IMPORTANT_PATHS):
+        python_utils.PRINT(
+            'scripts/pre_commit_linter.py --> Duplicate pattern(s) found '
+            'in CODEOWNER_IMPORTANT_PATHS list.')
+        failed = True
+
+    # Check missing rules by set difference operation.
+    critical_rule_section_minus_list_set = (
+        important_patterns_set.difference(codeowner_important_paths_set))
+    list_minus_critical_rule_section_set = (
+        codeowner_important_paths_set.difference(important_patterns_set))
+    for rule in critical_rule_section_minus_list_set:
+        python_utils.PRINT(
+            '%s --> Rule %s is not present in the '
+            'CODEOWNER_IMPORTANT_PATHS list in '
+            'scripts/pre_commit_linter.py. Please add this rule in the '
+            'mentioned list or remove this rule from the \'Critical files'
+            '\' section.' % (CODEOWNER_FILEPATH, rule))
+        failed = True
+    for rule in list_minus_critical_rule_section_set:
+        python_utils.PRINT(
+            '%s --> Rule \'%s\' is not present in the \'Critical files\' '
+            'section. Please place it under the \'Critical files\' '
+            'section since it is an important rule. Alternatively please '
+            'remove it from the \'CODEOWNER_IMPORTANT_PATHS\' list in '
+            'scripts/pre_commit_linter.py if it is no longer an '
+            'important rule.' % (CODEOWNER_FILEPATH, rule))
+        failed = True
+
+    return failed
+
+
+def check_codeowner_file(verbose_mode_enabled):
+    """Checks the CODEOWNERS file for any uncovered dirs/files and also
+    checks that every pattern in the CODEOWNERS file matches at least one
+    file/dir. Note that this checks the CODEOWNERS file according to the
+    glob patterns supported by Python2.7 environment. For more information
+    please refer https://docs.python.org/2/library/glob.html.
+    This function also ensures that the most important rules are at the
+    bottom of the CODEOWNERS file.
+    """
+    stdout = python_utils.string_io()
+    if verbose_mode_enabled:
+        python_utils.PRINT('Starting CODEOWNERS file check')
+        python_utils.PRINT('----------------------------------------')
+
+    with linter_utils.redirect_stdout(stdout):
```

Comment:
@kevinlee12 This method is not outputting anything, I do not be able to find a reason for that. Can you help me with it?